### PR TITLE
[BUGFIX] Processing R algorithms do not have enough outputs

### PR DESCRIPTION
**Pull request by rldhont**
R scripts in processing only supports Vector, Raster and Table.

This commit adds a way to get every outputs produced by an R script.

It is a revival of https://github.com/qgis/QGIS/commit/5bdd038e7422270b803687751f031d401f8df233

### CODE REVIEW

File: processing_r/processing/algorithm.py [hunk 1]
```diff
@@ -334,8 +340,56 @@ def processAlgorithm(self, parameters, context, feedback):
...
+        output2_value
+        """
+        if not self.save_output_values:
+            return
+
+        outputs = {}
+        output = None
+        ender = 0
+        line = next(lines).strip('\n').strip('\r')
+        while ender < 10:
```
Comment (nyalldawson) [inline processing_r/processing/algorithm.py]: Whats this supposed to do? break if there's 10 empty lines in a row? when would this happen?

File: processing_r/processing/algorithm.py [hunk 2]
```diff
@@ -291,6 +294,9 @@ def process_parameter_line(self, line):
                 self.error = self.tr('This script has a syntax error.\n'
                                      'Problem with line: {0}').format(line)
 
+        if self.save_output_values:
+            self.output_values_filename = QgsProcessingUtils.generateTempFilename('processing_values.txt')
```
Comment (nyalldawson) [inline processing_r/processing/algorithm.py]: Could you defer the filename generation to execute_r_algorithm ? We don't need to generate this name until we actually run the algorithm

**Comment by nyalldawson** [comment]
Thanks @rldhont , this looks very handy! I'll cut a new release as soon as we can get this merged

**Comment by nyalldawson** [comment]
@rldhont one last thing I forgot -- can you also include documentation updates please?  See https://github.com/north-road/qgis-processing-r/tree/master/website/pages

File: processing_r/processing/algorithm.py (hunk 1 above)
Comment (rldhont) [reply processing_r/processing/algorithm.py]: It is the same in `parse_script` and it was in the old plugin. The idea is to get multi-line values.

Some values can be multi-line and some line can be empty between 2 text lines.

File: processing_r/processing/algorithm.py (hunk 2 above)
Comment (rldhont) [reply processing_r/processing/algorithm.py]: I will do necessary.

File: processing_r/processing/algorithm.py (hunk 1 above)
Comment (nyalldawson) [reply processing_r/processing/algorithm.py]: Ok, fair enough

**Comment by rldhont** [comment]
Thanks @nyalldawson. I try to fix tests and I will update documentation.

**Comment by rldhont** [comment]
@nyalldawson:
* Test has been fixed
* Sample added
* Website updated

**Pull request merged**